File: start-server.js
#!/usr/bin/env node

const http = require('http');
const fs = require('fs');
const path = require('path');
const os = require('os');
const { exec } = require('child_process');

const PORT = 8000;

// Get local IP address
function getLocalIP() {
    const interfaces = os.networkInterfaces();
    for (const name of Object.keys(interfaces)) {
        for (const interface of interfaces[name]) {
            if (interface.family === 'IPv4' && !interface.internal) {
                return interface.address;
            }
        }
    }
    return 'localhost';
}

// MIME types
const mimeTypes = {
    '.html': 'text/html',
    '.js': 'text/javascript',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpg',
    '.gif': 'image/gif',
    '.svg': 'image/svg+xml',
    '.wav': 'audio/wav',
    '.mp4': 'video/mp4',
    '.woff': 'application/font-woff',
    '.ttf': 'application/font-ttf',
    '.eot': 'application/vnd.ms-fontobject',
    '.otf': 'application/font-otf',
    '.wasm': 'application/wasm'
};

const server = http.createServer((req, res) => {
    // Handle CORS
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    let filePath = '.' + req.url;
    if (filePath === './') {
        filePath = './index.html';
    }

    const extname = String(path.extname(filePath)).toLowerCase();
    const mimeType = mimeTypes[extname] || 'application/octet-stream';

    fs.readFile(filePath, (error, content) => {
        if (error) {
            if (error.code === 'ENOENT') {
                // File not found
                res.writeHead(404, { 'Content-Type': 'text/html' });
                res.end('<h1>404 Not Found</h1>', 'utf-8');
            } else {
                // Server error
                res.writeHead(500);
                res.end(`Server Error: ${error.code}`, 'utf-8');
            }
        } else {
            // Success
            res.writeHead(200, { 'Content-Type': mimeType });
            res.end(content, 'utf-8');
        }
    });
});

const localIP = getLocalIP();

console.log('='.repeat(60));
console.log('MCDO Website Server Started!');
console.log('='.repeat(60));
console.log(`Local access: http://localhost:${PORT}`);
console.log(`Phone access: http://${localIP}:${PORT}`);
console.log('='.repeat(60));
console.log('Instructions for phone access:');
console.log('1. Make sure your phone and computer are on the same WiFi network');
console.log('2. On your phone, open Chrome and go to the address above');
console.log('3. If it doesn\'t work, try these troubleshooting steps:');
console.log('   - Check firewall settings on your computer');
console.log('   - Make sure both devices are on the same network');
console.log('   - Try using the computer\'s IP address instead of hostname');
console.log('='.repeat(60));
console.log('Press Ctrl+C to stop the server');
console.log('='.repeat(60));

server.listen(PORT, '0.0.0.0', () => {
    console.log(`Server listening on port ${PORT}`);
    
    // Automatically open the browser
    const url = `http://localhost:${PORT}`;
    const startCommand = process.platform === 'darwin' ? 'open' : process.platform === 'win32' ? 'start' : 'xdg-open';
    exec(`${startCommand} ${url}`);
});

process.on('SIGINT', () => {
    console.log('\nServer stopped by user');
    process.exit(0);
});

File: start-server.py
#!/usr/bin/env python3
"""
Simple HTTP Server for MCDO Website
Run this script to start a local server accessible from your phone
"""

import http.server
import socketserver
import os
import sys
import webbrowser
from socket import gethostbyname, gethostname

# Change to the directory containing this script
os.chdir(os.path.dirname(os.path.abspath(__file__)))

# Set the port
PORT = 8000

# Get the local IP address
def get_local_ip():
    try:
        # Get the local IP address
        hostname = gethostname()
        local_ip = gethostbyname(hostname)
        return local_ip
    except:
        return "localhost"

# Create the server
Handler = http.server.SimpleHTTPRequestHandler

try:
    with socketserver.TCPServer(("", PORT), Handler) as httpd:
        local_ip = get_local_ip()
        
        print("=" * 60)
        print("MCDO Website Server Started!")
        print("=" * 60)
        print(f"Local access: http://localhost:{PORT}")
        print(f"Phone access: http://{local_ip}:{PORT}")
        print("=" * 60)
        print("Instructions for phone access:")
        print("1. Make sure your phone and computer are on the same WiFi network")
        print("2. On your phone, open Chrome and go to the address above")
        print("3. If it doesn't work, try these troubleshooting steps:")
        print("   - Check firewall settings on your computer")
        print("   - Make sure both devices are on the same network")
        print("   - Try using the computer's IP address instead of hostname")
        print("=" * 60)
        print("Press Ctrl+C to stop the server")
        
        # Automatically open the browser
        print(f"\nOpening http://localhost:{PORT} in your browser...")
        webbrowser.open(f"http://localhost:{PORT}")
        print("=" * 60)
        
        httpd.serve_forever()

except KeyboardInterrupt:
    print("\nServer stopped by user")
    sys.exit(0)
except Exception as e:
    print(f"Error starting server: {e}")
    print("\nTroubleshooting:")
    print("1. Make sure port 8000 is not in use")
    print("2. Check your firewall settings")
    print("3. Try running as administrator if needed")
    sys.exit(1)
